### PR TITLE
Add Doctor

### PR DESCRIPTION
**Pull request by Bryan-Goh**
Things added / changed:
Added the Doctor class
Added the UniqueDoctor class
Added Add and Delete Doctor Command
Edit Add Delete and Edit Appointment Command
Edit Appointment class
Edit the storage classes

### CODE REVIEW

File: src/main/java/seedu/address/model/doctor/Doctor.java
Comment (tohpinren) [inline src/main/java/seedu/address/model/doctor/Doctor.java]: Change this to name check only to follow isSameDoctor

File: src/main/java/seedu/address/logic/parser/CliSyntax.java
Comment (tohpinren) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java]: Can you change the prefix to something more intuitive? Like /doc

File: src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java
Comment (tohpinren) [inline src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java]: Create new duplicate message for duplicate doctors
Current message is "This appointment already exists for the patient."
Should be "This doctor already has an appointment at the same time."

File: src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java
Comment (tohpinren) [inline src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java]: Is it possible to change the editing of appointment's doctor to be by index? Since our add appointment command is by index also. Otherwise, write a message to tell users it is not by index.

File: src/main/java/seedu/address/storage/JsonAdaptedAppointment.java
Comment (tohpinren) [inline src/main/java/seedu/address/storage/JsonAdaptedAppointment.java]: Change field to doctorName to make it clearer

File: src/main/java/seedu/address/logic/commands/DeleteAppointmentCommand.java
Comment (tohpinren) [inline src/main/java/seedu/address/logic/commands/DeleteAppointmentCommand.java:76]: Is there a List function to find doctor in list by name?

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java
Comment (tohpinren) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java]: Same thing. If possible, there might be a better way to write this with in built functions

**Review by tohpinren — CHANGES_REQUESTED** [review]
Some comments to fix some issues.

Thanks!

**Review by simbayippy — COMMENTED** [review]
Nothing much to say apart from what was already said by @tohpinren. Very good work regardless, awesome pawsome!

File: src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java
Comment (simbayippy) [inline src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java]: can we change the error message displayed? currently if no doctors are available, error message thrown is "invalid patient Index"

"Invalid doctor index" would be a more suitable error message

File: src/main/java/seedu/address/logic/commands/DeleteAppointmentCommand.java
Comment (wj331) [reply src/main/java/seedu/address/logic/commands/DeleteAppointmentCommand.java:76]: Agree with this. Perhaps can use .get( ) method of List library

Comment (wj331) [reply src/main/java/seedu/address/logic/commands/DeleteAppointmentCommand.java:76]: Agree with this. Perhaps can use .get( ) method of List library

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java
Comment (wj331) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:68]: English in javadoc is incoherent? Maybe Doctor that has the appointment? minor nits

File: src/main/java/seedu/address/model/appointment/Appointment.java
Comment (wj331) [inline src/main/java/seedu/address/model/appointment/Appointment.java:43]: Can you be more specific with the "name"? Perhaps like nameOfDoctor or something. This makes me think appointment has a name on a first glance

File: src/main/java/seedu/address/storage/JsonAdaptedAppointment.java
Comment (wj331) [reply src/main/java/seedu/address/storage/JsonAdaptedAppointment.java]: yes agreed

**Review by wj331 — COMMENTED** [review]
LGTM except for some nits. Pls fix before merging

**Comment by yezkez10** [comment]
1 test still failing. Suggest to fix before the merge.
FindAppointmentCommandTest > execute_multipleKeywords_multipleAppointmentsFound() FAILED
    org.opentest4j.AssertionFailedError at FindAppointmentCommandTest.java:77
> Task :test FAILED

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java
Comment (yezkez10) [reply src/main/java/seedu/address/logic/commands/DeleteCommand.java:68]: Suggest: Returns the Doctor that is in charge of this appointment

File: src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java
Comment (yezkez10) [inline src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java:108]: Needs more documentation. 
Suggest: 
Returns the Doctor in the doctorList with the given Name. 
param: doctorList the list of all Doctors
param: doctorName the name of the Doctor being searched for

**Review by yezkez10 — APPROVED** [review]
Approving so that can merge and continue frontend

**Review by tohpinren — APPROVED** [review]

**Pull request merged**
